appium/webdriver/appium_connection.py

```diff
@@ -26,55 +25,40 @@
 
 PREFIX_HEADER = 'appium/'
 
+HEADER_IDEMOTENCY_KEY = 'X-Idempotency-Key'
 
-class AppiumConnection(RemoteConnection):
-    _proxy_url: Optional[str]
-
-    def __init__(
-        self,
-        remote_server_addr: str,
-        keep_alive: bool = False,
-        ignore_proxy: Optional[bool] = False,
-        init_args_for_pool_manager: Union[Dict[str, Any], None] = None,
-    ):
-        # Need to call before super().__init__ in order to pass arguments for the pool manager in the super.
-        self._init_args_for_pool_manager = init_args_for_pool_manager or {}
-
-        super().__init__(remote_server_addr, keep_alive=keep_alive, ignore_proxy=ignore_proxy)
-
-    def _get_connection_manager(self) -> Union[urllib3.PoolManager, urllib3.ProxyManager]:
-        # https://github.com/SeleniumHQ/selenium/blob/0e0194b0e52a34e7df4b841f1ed74506beea5c3e/py/selenium/webdriver/remote/remote_connection.py#L134
-        pool_manager_init_args = {'timeout': self.get_timeout()}
-
-        if self._ca_certs:
-            pool_manager_init_args['cert_reqs'] = 'CERT_REQUIRED'
-            pool_manager_init_args['ca_certs'] = self._ca_certs
-        else:
-            # This line is necessary to disable certificate verification
-            pool_manager_init_args['cert_reqs'] = 'CERT_NONE'
 
-        pool_manager_init_args.update(self._init_args_for_pool_manager)
+def _get_new_headers(key: str, headers: Dict[str, str]) -> Dict[str, str]:
+    """Return a new dictionary of heafers without the given key.
+    The key match is case-insensitive."""
+    key = key.lower()
```

Comment:
I usually try to avoid reassigning of method/function arguments

Reply:
yea. done